internal/k8s/controller.go

```diff
@@ -141,14 +141,18 @@ func (k *Kubernetes) SetStore(store store.Store) {
 	k.store = store
 }
 
+func (k *Kubernetes) Client() gatewayclient.Client {
```

Comment:
We need external access to the `gatewayclient` after constructing the controller in `internal/commands/server/server.go`; otherwise, we have to duplicate a bunch of kube client init logic in both places.